### PR TITLE
Replace Pdocs  with Mkdocstrings

### PR DESCRIPTION
**Pull request by hotzenklotz**
### Description:
This PR replaces our pieced-together, hotglue API documentation build step.
- Replace `pdoc`with [`mkdocstrings`](https://mkdocstrings.github.io/)
  - Directly compatible with `mkdocs` and it's styling
  - Remove custom templates and bash scripts
  - Build time dramatically improves from 30s+ to 5s and live reloading
  
Preview:
Screenshot 2024-08-22 at 10 42 58

### Testing
- `cd ./docs`
- `./generate.sh`
- Open browser on page `http://localhost:8197` and check API docs, e.g. http://localhost:8197/api/webknossos/dataset/layer.html

### Issues:
- None

### Todos:
Make sure to delete unnecessary points or to check all before merging:
 - [ ] Updated Changelog
 - [x] Updated Documentation
 - [ ] Added / Updated Tests
 - [ ] Considered adding this to the Examples

### CODE REVIEW

**Review by markbader — CHANGES_REQUESTED** [review]
Great work :tada: I like the appearance of the docs and the build time. It seems as if there are quite a few links broken, due to changes between pdoc and mkdocs. I started to fix obvious issues, but there are still some broken links left (see mkdocs warnings) where I couldn't find the destination easily.

File: docs/mkdocs.yml [hunk 1]
```diff
@@ -115,23 +115,23 @@ nav:
       - API Reference:
           - Overview: api/webknossos.md
```
[the remark this answers is not in the document]
Comment (hotzenklotz) [reply docs/mkdocs.yml:116]: Good catch. I want to manually create that. Let me check.

**Comment by hotzenklotz** [comment]
> I started to fix obvious issues, but there are still some broken links left (see mkdocs warnings) where I couldn't find the destination easily.

I will fix all links in PR #1171 , where we will reorganize the docs anyway. Perhaps we do not publish this PR/docs until then.

**Review by markbader — COMMENTED** [review]
Okay, great! Another thing I noticed when updating markdown files during the development server runs, there seems to be an infinite update loop, so just when the docs are build again the -something-has-changed- trigger fires again. I am not sure if this is some misconfiguration on my side or an issue with mkdocs. Maybe you can check again on your machine.

File: docs/mkdocs.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (hotzenklotz) [reply docs/mkdocs.yml:116]: Fixed

**Pull request merged**